src/clusterfuzz/_internal/build_management/build_manager.py

```diff
@@ -764,22 +765,71 @@ def setup(self):
+
+  - a symbolized release build
+  - a symbolized debug build
 
-  def __init__(self, base_build_dir, revision, release_build_url,
-               debug_build_url):
+  Members:
+    release_build_url (Optional[str]): The URL from which the release build is
+      downloaded, if any.
+    debug_build_url (Optional[str]): The URL from which the debug build is
+      downloaded, if any.
+    reuse_release_build (bool): Whether the release build is re-used from an
+      existing build instead of downloaded anew.
+  """
+
+  def __init__(self,
+               base_build_dir: str,
+               revision: int,
+               release_build_url: Optional[str],
+               debug_build_url: str,
+               existing_build: Optional[Build] = None):
+    """Initializes a symbolized build.
+
+    Args:
+      base_build_dir: The directory in which to store symbolized builds.
+      revision: The revision at which to download symbolized builds.
+      release_build_url: The URL from which to download the release build, if
+        any.
+      debug_build_url: The URL from which to download the debug build, if any.
+      existing_build: An existing build to consider re-using as the symbolized
+        release build. This is reused iff its base build directory is the same
+        as this instance's, its URL is the same as `release_build_url` and its
+        revision is the same as `revision`. In that case,
+        `self.release_build_dir` is set to that build's directory, which is not
+        under `self.build_dir`.
+    """
     super().__init__(base_build_dir, revision)
     self._build_dir = os.path.join(self.base_build_dir, 'symbolized')
-    self.release_build_dir = os.path.join(self.build_dir, 'release')
-    self.debug_build_dir = os.path.join(self.build_dir, 'debug')
 
     self.release_build_url = release_build_url
     self.debug_build_url = debug_build_url
 
+    # If we have already set up the same build as an existing build, re-use it.
+    # We only re-use regular builds, which are downloaded from GCS like
+    # symbolized builds are.
+    self._reuse_release_build = (
+        existing_build and isinstance(existing_build, RegularBuild) and
+        self.base_build_dir == existing_build.base_build_dir and
+        self.release_build_url == existing_build.build_url and
+        self.revision == existing_build.revision)
+    if self._reuse_release_build:
+      logs.log(f'Re-using symbolized release build: {existing_build.build_dir}')
+      self.release_build_dir = existing_build.build_dir
+    else:
+      self.release_build_dir = os.path.join(self.build_dir, 'release')
+
+    self.debug_build_dir = os.path.join(self.build_dir, 'debug')
+
   @property
   def build_dir(self):
     return self._build_dir
 
+  @property
+  def reuse_release_build(self):
+    return self._reuse_release_build
+
   def _unpack_builds(self):
     """Download and unpack builds."""
     if not shell.remove_directory(self.build_dir, recreate=True):
```

Comment:
Wouldn't this delete the existing symbolized build (if any)?

Reply:
My bad, self.build_dir is a function call that returns `_build_dir`. The existing build is not present under this directory.